rocketmq-console/src/main/java/org/apache/rocketmq/console/service/impl/MessageTraceServiceImpl.java

```diff
@@ -42,11 +42,9 @@
     @Override public List<MessageTraceView> queryMessageTraceByTopicAndKey(String topic, String key) {
         try {
             List<MessageTraceView> messageTraceViews = new ArrayList<MessageTraceView>();
-            List<MessageExt> messageTraceList = mqAdminExt.queryMessage(topic, key, QUERY_MESSAGE_MAX_NUM, 0, System.currentTimeMillis()).getMessageList();
-            for (MessageExt messageExt : messageTraceList) {
-                List<MessageTraceView> messageTraceView = MessageTraceView.decodeFromTraceTransData(key, new String(messageExt.getBody(), Charsets.UTF_8));
-                messageTraceViews.addAll(messageTraceView);
-            }
+            MessageExt messageExt = mqAdminExt.viewMessage(topic, key);
```

Comment:
Do you omit the range of the query?

Reply:
QueryMessage method seems have a different parameter with ViewMessage method, and  QueryMessage need a input parameter **key**, while here we query by msgId, so always the api return "query message by key finished, but no message".
Actually, I'm not quite sure about the difference between **key** and **msgId**, so I read the source code of **QueryMessageProcessor** class , it has two methods viewMessageById and queryMessage, queryMessage seems a kind of batch search.

Reply:
In this case, you could use a key to trace messages with the same key.

Reply:
I've figured it out. I used the wrong topic **RMQ_SYS_TRACE_TOPIC**. Thanks!